src/main/java/teammates/logic/core/AccountsLogic.java

```diff
@@ -162,6 +164,16 @@ public InstructorAttributes joinCourseForInstructor(String encryptedKey, String
         return instructor;
     }
 
+    private void validateInstructorInstitute (InstructorAttributes instructor, String institute)
+            throws InvalidParametersException {
+        assert instructor != null : "Should have been checked in validateInstructorJoinRequest() method.";
+        AccountAttributes account = getAccount(instructor.email);
```

Comment:
There are two problems here:
- `getAccount` accepts `googleId` rather than `email`
- If instructor does not have account stored in the database (e.g. it is the first time he/she use TEAMMATES), account will be null. In this case, they can still modify the institute name and bypass the check.

Reply:
Apologies for the mistake relating to the parameter for `getAccount`.

If accounts are identified by their googleId, how does the [CreateAccountAction](https://github.com/TEAMMATES/teammates/blob/8160fa8f0fc07c918eb3b243024014dfe603adaf/src/main/java/teammates/ui/webapi/action/CreateAccountAction.java#L43) create an account (given that no google id is provided)?

If that class doesn't actually create an account, then how are instructors loaded in the admin home page (for admins to add instructors/generate the join link)? Unless that data is never stored and merely persisted in the live server?

Appreciate any insight you can provide, thank you for your time!

Reply:
> If accounts are identified by their googleId, how does the CreateAccountAction create an account (given that no google id is provided)?

:p apologies for the naming, it actually does not create an account.

![image](https://user-images.githubusercontent.com/13977235/72317414-09eb8380-364e-11ea-85d1-4e798ee304a1.png)

The action will create everything except for `Account` and `StudentProfile`.

> If that class doesn't actually create an account, then how are instructors loaded in the admin home page

When `Instructor` joins course, we store its googleId in the `Instructor` entity (the instructor is get by the `encrytedKey`).

Reply:
I see, thank you so much for the information! 

I will think on this problem for a little longer. Should I close this PR and open a new one when I have a solution or just push updates to this PR?

Reply:
Let's close it right now :)

I think approaching the problem without changing the data schema is a bit difficult. I suggest you switch to/ or you can find your own issue to work on.

Reply:
Understood, thanks again for the help :)